src/pages/settings/Profile/Contacts/ContactMethodsPage.js

```diff
@@ -0,0 +1,138 @@
+import Str from 'expensify-common/lib/str';
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {ScrollView} from 'react-native-gesture-handler';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import HeaderWithCloseButton from '../../../../components/HeaderWithCloseButton';
+import ScreenWrapper from '../../../../components/ScreenWrapper';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '../../../../components/withCurrentUserPersonalDetails';
+import withLocalize, {withLocalizePropTypes} from '../../../../components/withLocalize';
+import CONST from '../../../../CONST';
+import compose from '../../../../libs/compose';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import ROUTES from '../../../../ROUTES';
+import LoginField from './LoginField';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** Value of partner name */
+        partnerName: PropTypes.string,
+
+        /** Phone/Email associated with user */
+        partnerUserID: PropTypes.string,
+
+        /** Date of when login was validated */
+        validatedDate: PropTypes.string,
+    }),
+
+    ...withLocalizePropTypes,
+    ...withCurrentUserPersonalDetailsPropTypes,
+};
+
+const defaultProps = {
+    loginList: {},
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+
+class ContactMethodsPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            logins: this.getLogins(),
+        };
```

Comment:
Do you know why we copy this to a state instead of just directly calling `this.getLogins()` on render and use what we get there? synchronizing the prop with the state adds more complexity and in this case doesn't seem to serve any purpose.

Reply:
Def agreed - i have no idea why we did this in the first place, and if we want to take the time to refactor I def agree it makes sense to just directly use what we get from props and not deal with syncing it with state in `componentDidUpdate` 👍